pommerman/envs/v3.py

```diff
@@ -0,0 +1,101 @@
+"""The Pommerman v3 Environment, which implements Sudden Death mode.
+
+The game ends when any agent dies. In FFA, all other agents win. In any of the team
+modes, the team with the most remaining players wins. If all players die at the same
+time or the same number of players die on each team, the game ends in a tie. All
+players/teams lose if the game ends without anyone dying.
+"""
+from .. import constants
+from .. import forward_model
+from .. import utility
+from . import v0
+
+
+class Pomme(v0.Pomme):
+    metadata = {
+        'render.modes': ['human', 'rgb_array']
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _get_done(self):
+        alive = [agent for agent in self._agents if agent.is_alive]
+        alive_ids = sorted([agent.agent_id for agent in alive])
+        if self._step_count >= self._max_steps:
+            return True
+        elif len(alive) < len([x for x in self._agents if x._agent_type != 'dummy']):
+            return True
+        return False
+
+    def _get_info(self, done, rewards):
```

Comment:
Same as above. Don't skip using the forward_model. Change the result instead to be specific to SuddenDeath only. That way, we don't break the abstraction layer.